### PR TITLE
docs: Add RabbitMQ module documentation incl. an example

### PR DESCRIPTION
**Pull request by annaerdi**
## What does this PR do?

Adds module documentation page for RabbitMQ testcontainer

## Why is it important?

The PR is contributing with one module-specific documentation, which was not available before.

## Related issues

-

### CODE REVIEW

File: docs/modules/rabbitmq.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        var body = eventArgs.Body.ToArray();
+        receivedMessage = Encoding.UTF8.GetString(body);
+    };
+
+    channel.BasicConsume(queueName, true, consumer);
+    Thread.Sleep(1000); // to avoid race condition
+    Assert.Equal(message, receivedMessage);
+  }
+
+}
```
Comment (HofmeisterAn) [inline docs/modules/rabbitmq.md]:
```suggestion
using System.Text;
using System.Threading;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;
using Testcontainers.RabbitMq;
using Xunit;

public sealed class RabbitMqContainerTest : IAsyncLifetime
{
  private readonly RabbitMqContainer _rabbitMqContainer = new RabbitMqBuilder().Build();

  public Task InitializeAsync()
  {
    return _rabbitMqContainer.StartAsync();
  }

  public Task DisposeAsync()
  {
    return _rabbitMqContainer.DisposeAsync().AsTask();
  }

  [Fact]
  public void ConsumeMessageFromQueue()
  {
    const string queue = "hello";

    const string message = "Hello World!";

    string actualMessage = null;

    // Signal the completion of message reception.
    EventWaitHandle waitHandle = new ManualResetEvent(false);

    // Create and establish a connection.
    var connectionFactory = new ConnectionFactory();
    connectionFactory.Uri = new Uri(_rabbitMqContainer.GetConnectionString());
    using var connection = connectionFactory.CreateConnection();

    // Send a message to the channel.
    using var channel = connection.CreateModel();
    channel.QueueDeclare(queue, false, false, false, null);
    channel.BasicPublish(string.Empty, queue, null, Encoding.Default.GetBytes(message));

    // Consume a message from the channel.
    var consumer = new EventingBasicConsumer(channel);
    consumer.Received += (_, eventArgs) =>
    {
      actualMessage = Encoding.Default.GetString(eventArgs.Body.ToArray());
      waitHandle.Set();
    };

    channel.BasicConsume(queue, true, consumer);
    waitHandle.WaitOne(TimeSpan.FromSeconds(1));

    Assert.Equal(message, actualMessage);
  }
}
```

Basically, I replaced the thread sleep with a semaphore.

File: docs/modules/rabbitmq.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# RabbitMQ
+
+Here is an example of a pre-configured RabbitMQ [module](https://www.nuget.org/packages/Testcontainers.RabbitMq;).
```
Comment (HofmeisterAn) [inline docs/modules/rabbitmq.md]:
```suggestion
Here is an example of a pre-configured RabbitMQ [module](https://www.nuget.org/packages/Testcontainers.RabbitMq).
```

File: docs/modules/rabbitmq.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    channel.BasicConsume(queueName, true, consumer);
+    Thread.Sleep(1000); // to avoid race condition
+    Assert.Equal(message, receivedMessage);
+  }
+
+}
+```
+
+Make sure to install the [RabbitMQ.Client](https://www.nuget.org/packages/RabbitMQ.Client) package to be able to connect to RabbitMQ.
```
Comment (HofmeisterAn) [inline docs/modules/rabbitmq.md]: Should we add the required dependencies (as copy and paste snippet) to the top of the example, e.g. something like:

~~~
```console title="Install the NuGet dependencies"
dotnet add package Testcontainers.RabbitMq
dotnet add package RabbitMQ.Client
```
~~~

and then

~~~
```csharp title="Run RabbitMQ test"
```
~~~

?

**Review by HofmeisterAn — CHANGES_REQUESTED** [review]
Thank you for submitting your pull request! I have a few small suggestions. What are your thoughts on them?

**Comment by annaerdi** [comment]
Thank you for the suggestions, I modified it accordingly

**Review by HofmeisterAn — APPROVED** [review]

**Comment by HofmeisterAn** [comment]
> Thank you for the suggestions, I modified it accordingly

Thanks for your contribution!

**Pull request merged**